src/main/java/com/auction/auction/bid/service/UserService.java

```diff
@@ -2,62 +2,76 @@
 
 import com.auction.auction.bid.exception.EmailAlreadyInUseException;
 import com.auction.auction.bid.exception.UserNotFoundException;
+import com.auction.auction.bid.interfaces.UserInterface;
 import com.auction.auction.bid.model.User;
 import com.auction.auction.bid.repository.UserRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.stereotype.Service;
+import com.auction.auction.bid.dto.UserDTO;
 
 import java.util.List;
 
 @Service
-public class UserService {
+public class UserService implements UserInterface {
 
     @Autowired
-    private final UserRepository userRepository;
-    private final BCryptPasswordEncoder passwordEncoder;
+    private UserRepository userRepository;
 
-    public UserService(UserRepository userRepository, BCryptPasswordEncoder passwordEncoder) {
-        this.userRepository = userRepository;
-        this.passwordEncoder = passwordEncoder;
-    }
+    @Autowired
+    private BCryptPasswordEncoder passwordEncoder;
 
-    public User createUser(User user) {
-        if (userRepository.existsByEmail(user.getEmail())) {
-            throw new EmailAlreadyInUseException(user.getEmail());
-        }
-        user.setPassword(passwordEncoder.encode(user.getPassword()));
+    public User saveUser(User user) {
         return userRepository.save(user);
     }
 
-    public User saveOrUpdateUser(User user) {
-        return userRepository.save(user);
+    @Override
+    public User createUser(UserDTO userDTO) {
+        if (userRepository.existsByEmail(userDTO.getEmail())) {
+            throw new EmailAlreadyInUseException("Email is already in use: " + userDTO.getEmail());
+        }
+
+        User user = new User();
+        user.setUsername(userDTO.getUsername());
+        user.setPassword(passwordEncoder.encode(userDTO.getPassword()));
+        user.setEmail(userDTO.getEmail());
+        return saveUser(user);
     }
 
+    @Override
     public User getUserById(String id) {
         return userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(String.format("User with id %s not found", id)));
     }
 
+    @Override
     public List<User> getAllUsers() {
         return userRepository.findAll();
     }
 
-    public boolean existsByEmail(String email) {
-        return userRepository.existsByEmail(email);
-    }
-
+    @Override
     public List<User> getUserByUsername(String username) {
         return userRepository.findByUsername(username);
     }
-    public List<User> getUserByEmail(String email) {
-        return userRepository.findByEmail(email);
+
+    @Override
+    public User updateUser(String id, User user) {
+        User existingUser = userRepository.findById(id)
```

Comment:
here you can reuse getUserById() method because it already contains appropriate exception in it